src/constraint/encoding.ts

```diff
@@ -239,6 +239,28 @@ export const ENCODING_CONSTRAINTS: EncodingConstraintModel[] = [
       return true;
     }
   },{
+    name: 'scalePropertiesSupportedByChannel',
+    description: 'Not all scale properties are supported by all encoding channels',
+    properties: [Property.CHANNEL, Property.SCALE],
+    allowWildcardForProperties: false, // unsure about this
+    strict: true,
+    satisfy: (encQ: EncodingQuery) => {
+      if (encQ) {
+        let channel = encQ.channel as Channel;
+        let scale = encQ.scale;
+        if (channel && scale) {
+          let scaleProps = Object.keys(scale);
+          for (let scaleProp in scaleProps) {
```

Comment:
`scaleProp in scale` is fine too.  

But you probably wanna use `scale.hasOwnProperty()` check like typical object key loop in JS.
More importantly, check if the prop is actually scale property (not wildcard's `name` and `enum`)